### PR TITLE
Find device before create

### PR DESCRIPTION
**Pull request by hmallat**

### CODE REVIEW

**Comment by amccarthy** [comment]
I don't see a watcher->deleteLater() call in the slots connected to QDBusPendingCallWatcher. This is a memory leak and which seems to be pre-existing. Please add a deleteLater() call.

Otherwise LGTM.

**Comment by hmallat** [comment]
watcher deleteLater() added

**Comment by amccarthy** [comment]
LGTM

**Comment by rburchell** [comment]
Upstream reviews:

https://codereview.qt-project.org/#/c/123808/
https://codereview.qt-project.org/#/c/123809/
https://codereview.qt-project.org/#/c/123810/

File: src/bluetooth/qbluetoothservicediscoveryagent_bluez.cpp [hunk 1]
```diff
@@ -130,45 +129,75 @@ void QBluetoothServiceDiscoveryAgentPrivate::stop()
...
+    qCDebug(QT_BT_BLUEZ) << Q_FUNC_INFO << "created" << address.toString();
+
+    QDBusPendingReply<QDBusObjectPath> deviceObjectPath = *watcher;
+    watcher->deleteLater();
+    if (deviceObjectPath.isError()) {
+        if (deviceObjectPath.error().name() != QLatin1String("org.bluez.Error.AlreadyExists")) {
+            qCDebug(QT_BT_BLUEZ) << "Create device failed Error: " << error << deviceObjectPath.error().name();
             delete adapter;
             adapter = 0;
-            if (singleDevice) {
```
Comment (rburchell) [inline src/bluetooth/qbluetoothservicediscoveryagent_bluez.cpp:159]: As pointed out on IRC, this removal isn't quite right. Error state needs to be reported up.

**Comment by hmallat** [comment]
Added error reporting to failing FindDevice/CreateDevice calls

**Pull request merged**